Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java

```diff
@@ -141,18 +139,34 @@ protected void addCompletions(@NotNull final CompletionParameters parameters,
                  }
 
                  updatedResultSet.addElement(lookupElement);
-               }, (includedSet, includedKinds, includedRelevanceTags) -> {
+               }, (includedSet, includedKinds, includedRelevanceTags, libraryFile) -> {
                  final AvailableSuggestionSet suggestionSet = das.getAvailableSuggestionSet(includedSet.getId());
                  if (suggestionSet == null) {
                    return;
                  }
 
+                 final Map<String, Map<String, Set<String>>> existingImports = das.getExistingImports(libraryFile);
                  for (AvailableSuggestion suggestion : suggestionSet.getItems()) {
                    final String kind = suggestion.getElement().getKind();
                    if (!includedKinds.contains(kind)) {
                      continue;
                    }
 
+                   Set<String> importedLibraries = new HashSet<>();
+                   for (Map.Entry<String, Map<String, Set<String>>> entry : existingImports.entrySet()) {
```

Comment:
@lambdabaa, just to check the environment: do you get a warning here?

![image](https://user-images.githubusercontent.com/4499789/59203956-53e82480-8ba8-11e9-8975-ae761885fdc1.png)

And IntelliJ IDEA is completely right. You get NPEs with Dart SDK 1.23.1 or older.

Reply:
Yes, I see it now. I went ahead and added a safety check for null existingImports. Thanks!